src/hotspot/share/interpreter/linkResolver.cpp

```diff
@@ -806,6 +823,8 @@ Method* LinkResolver::resolve_method(const LinkInfo& link_info,
   if (link_info.check_loader_constraints()) {
     // check loader constraints
     check_method_loader_constraints(link_info, resolved_method, "method", CHECK_NULL);
+
+    check_restricted_method(link_info, resolved_method, CHECK_NULL);
```

Comment:
Not at all clear why this is guarded by `check_loader_constraints`. ?? Otherwise the placement of the check seems fine.